html/includes/authenticate.inc.php

```diff
@@ -56,9 +56,13 @@
                 $username = clean($_REQUEST['username']);
                 $password = $_REQUEST['password'];
             } elseif (isset($_SERVER['REMOTE_USER'])) {
-                $username = $_SERVER['REMOTE_USER'];
-            } elseif (isset($_SERVER['PHP_AUTH_USER']) && $config['auth_mechanism'] === 'http-auth') {
-                $username = $_SERVER['PHP_AUTH_USER'];
+                $username = clean($_SERVER['REMOTE_USER']);
+            } elseif ($config['auth_mechanism'] === 'http-auth') {
+                if (isset($_SERVER['PHP_AUTH_USER']) && $_SERVER['PHP_AUTH_USER']) {
+                    $username = clean($_SERVER['PHP_AUTH_USER']);
+                } elseif (isset($config['http_auth_guest'])) {
```

Comment:
Would this elseif check even trigger. PHP_AUTH_USER should always contain someone who is authenticated.